yb-voyager/src/tgtdb/suites/oracleSuite.go

```diff
@@ -0,0 +1,206 @@
+		}
+		return formattedTimestamp, nil
+	},
+	"io.debezium.time.NanoTimestamp": func(columnValue string, formatIfRequired bool) (string, error) {
+		epochNanoSecs, err := strconv.ParseInt(columnValue, 10, 64)
+		if err != nil {
+			return "", fmt.Errorf("parsing epoch nanoseconds: %v", err)
+		}
+		epochSeconds := epochNanoSecs / 1000000000
+		epochNanos := epochNanoSecs % 1000000000
+		nanoTimestamp := time.Unix(epochSeconds, epochNanos).Local()
+		oracleTimestampFormat := "02-01-2006 03.04.05.000000000 PM" //format: DD-MON-YYYY HH.MI.SS.FFFFFFFFF PM
+		formattedTimestamp := nanoTimestamp.Format(oracleTimestampFormat)
+		if formatIfRequired {
+			formattedTimestamp = fmt.Sprintf("'%s'", formattedTimestamp)
+		}
+		return formattedTimestamp, nil
+	},
+	"io.debezium.time.ZonedTimestamp": func(columnValue string, formatIfRequired bool) (string, error) {
+		debeziumFormat := "2006-01-02T15:04:05Z07:00"
+
+		parsedTime, err := time.Parse(debeziumFormat, columnValue)
+		if err != nil {
+			return "", fmt.Errorf("parsing timestamp: %v", err)
+		}
+
+		oracleFormat := "06-01-02 3:04:05.000000000 PM -07:00"
+		if parsedTime.Location().String() == "UTC" { //LOCAL TIMEZONE Case
+			oracleFormat = "06-01-02 3:04:05.000000000 PM" // TODO: for timezone ones sqlldr is inserting as GMT though GMT/UTC is similar
+		}
+		formattedTimestamp := parsedTime.Format(oracleFormat)
+		if formatIfRequired {
+			formattedTimestamp = fmt.Sprintf("'%s'", formattedTimestamp)
+		}
+		return formattedTimestamp, nil
+	},
+	"BYTES": func(columnValue string, formatIfRequired bool) (string, error) {
+		//decode base64 string to bytes
+		decodedBytes, err := base64.StdEncoding.DecodeString(columnValue) //e.g.`////wv==` -> `[]byte{0x00, 0x00, 0x00, 0x00}`
+		if err != nil {
+			return columnValue, fmt.Errorf("decoding base64 string: %v", err)
+		}
+		//convert bytes to hex string e.g. `[]byte{0x00, 0x00, 0x00, 0x00}` -> `0000000`
+		hexString := ""
+		for _, b := range decodedBytes {
+			hexString += fmt.Sprintf("%02x", b)
+		}
+		hexValue := hexString
+		if formatIfRequired {
+			hexValue = fmt.Sprintf("'%s'", hexString) // in insert statement no need of escaping the backslash and add quotes
+		}
+		return string(hexValue), nil
+	},
+	"MAP": func(columnValue string, _ bool) (string, error) {
+		mapValue := make(map[string]interface{})
+		err := json.Unmarshal([]byte(columnValue), &mapValue)
+		if err != nil {
+			return columnValue, fmt.Errorf("parsing map: %v", err)
+		}
+		var transformedMapValue string
+		for key, value := range mapValue {
+			transformedMapValue = transformedMapValue + fmt.Sprintf("\"%s\"=>\"%s\",", key, value)
+		}
+		return fmt.Sprintf("'%s'", transformedMapValue[:len(transformedMapValue)-1]), nil //remove last comma and add quotes
+	},
+	"STRING": func(columnValue string, formatIfRequired bool) (string, error) {
+		if formatIfRequired {
+			formattedColumnValue := strings.Replace(columnValue, "'", "''", -1)
```

Comment:
I think we got rid of this? Just be careful when merging

Reply:
no, we are still formatting the values. It's just for prepared statements I am not formatting these values by unsettling the `formatIfRequired`.